### ISSUE TITLE
use groupname to set the acl

### ISSUE THREAD

**Issue by astream2018** [kellobri/azure-devops-shiny-template#4]
@kellobri 
Currently i set the acl based on the GROUP_GUID. I would like first to get the group name from the yaml as value do a get groups call get the GROUP_GUID based on the name and then call this script to set the acl script below.

#!/usr/bin/env bash
#
# Update the ACL settings for a piece of content on RSC
# given the Content GUID and a valid API key
#

# Build the JSON request data for adding a Viewer Group to the content ACL
DATA=$(jq --arg principal_guid "${GROUP_GUID}" \
   '. | .["principal_guid"]=$principal_guid' \
   <<<'{"principal_type":"group","role":"owner"}')

echo "${DATA}"

RESULT=$(curl --silent --show-error -L --max-redirs 0 --fail -X POST \
    -H "Authorization: Key ${CONNECT_API_KEY}" \
    --data-binary "${DATA}" \
    "${CONNECT_SERVER}__api__/v1/content/${CONTENT}/permissions")
RESPONSE=$(echo "$RESULT" | jq -r .id)

echo "ACL permissions: ${RESPONSE} Update Complete."r

**Comment by kellobri** [comment]
We don't currently have an API for making that request. The closest we could do today would be enumerate all the groups, then filter them to return just the information relevant to the group you're looking for. That's rather heavy-handed though, and involves pagination. It also looks like the groups enumeration endpoint may only return local groups. I can put this down as a feature request for sure. Is this a big blocker for you right now?

**Comment by Mohamed-0-Hassan** [comment]
Hi @kellobri 
Maybe I didn't write it quite clear. 
Add parameter i would like to give the group name.
I would like to get the guid for the group name to execute the call above.

Currently users provide the GUID, it would better if they could provide the name.

**Comment by kellobri** [comment]
Yes, that's what I understood you were asking for. We don't currently have an API for making that form of the request.

**Comment by astream2018** [comment]
@kellobri 
What you could do is, when the group name is provided in the yaml as value.
Do a get all the groups (__api__/v1/groups) and compare the name against the list and return the  guid of the group (${GROUP_GUID) .

This could be the input value of the call above.

**Comment by astream2018** [comment]
@kellobri
What you could do is, when the group name is provided in the yaml as value.
Do a get all the groups (api/v1/groups) and compare the name against the list and return the guid of the group (${GROUP_GUID) .

This could be the input value of the call above.
